### DISCUSSION TITLE
ST25R3916/src /rfal_rfst25r3916.h

### DISCUSSION THREAD

**Discussion by Lemac13** [stm32duino/Arduino_Core_STM32#2430] [Libraries]
Hello 

In the Line 51 of the file "[ST25R3916](https://github.com/stm32duino/ST25R3916/tree/main)/[src](https://github.com/stm32duino/ST25R3916/tree/main/src)
/rfal_rfst25r3916.h"

You have a inclued
#include <functional>

This file is no in the library and it crass the compilation

I try to commnet it. But it crass in other point

BR

**Comment by fpistm** [comment]
Which core you used?

**Reply by Lemac13** [reply]
I am using a Arduino leonardo or a Arduino uno R4 Wifi

**Comment by fpistm** [comment]
I don't think library is compatible. @cparata ?

**Reply by Lemac13** [reply]
Which core must i used?

I am usign a EVB "X-NUCLEO-NFC08A1" with a Arduino uno R4 Wifi

![image](https://github.com/stm32duino/Arduino_Core_STM32/assets/20709388/3f54f779-8052-4907-8515-958f00e2acba)

Can I not use a Arduino board??

**Reply by fpistm** [reply]
This is the connector. Nucleo boards have it. Unfortunately I guess the functional.h (guess this is the include which cause the issue as your post is not precise on the include...) is not provided/ compatible with arduino uno core.

**Reply by Lemac13** [reply]
Which core must i use?

**Reply by fpistm** [reply]
The STM32 core: https://github.com/stm32duino/Arduino_Core_STM32
But you have to use a stm32 based board.
